test/core/utils/preload.js

```diff
@@ -0,0 +1,140 @@
+describe('axe.utils.preload unit tests', function() {
+	'use strict';
+
+	describe('axe.utils.preload', function() {
+		it('should be a function', function() {
+			assert.isFunction(axe.utils.preload);
+		});
+
+		it('should return a queue', function() {
+			var options = {
+				preload: true
+			};
+			var actual = axe.utils.preload(options);
+			assert.isObject(actual);
+		});
+
+		it('should ensure queue is defer(able)', function(done) {
+			var options = {
+				preload: false
+			};
+			var actual = axe.utils.preload(options);
+			actual
+				.defer(function(res, rej) {
+					assert.isFunction(rej);
+					res(true);
+					done();
+				})
+				.catch(function(error) {
+					done(error);
+				});
+		});
+
+		it('should ensure queue is then(able)', function(done) {
+			var options = {
+				preload: false
+			};
+			var actual = axe.utils.preload(options);
+			actual
+				.then(function(results) {
+					assert.isDefined(results);
+					done();
+				})
+				.catch(function(error) {
+					done(error);
+				});
+		});
+
+		it('should return an object with property cssom', function(done) {
```

Comment:
This should test that the CSS preload function is called correctly, and that the returned value is the one that the CSS preload returned.

Reply:
Sorted!